information/specification.md

```diff
@@ -0,0 +1,706 @@
+    **_t_**.
+
+### Transaction Output
+
+A _transaction output_ is a pair of values (**_t_**, **_a_**), where:
+
+  * **_t_** is a [target address](#address);
+  * **_a_** is a [coin amount](#coin-amount).
+
+# Interface
+
+All coin selection algorithms used by Cardano Wallet implement a _common
+interface_.
+
+Essentially, a coin selection algorithm is simply a _mathematical function_
+that, when applied to a [standard set of parameters](#parameters), will produce
+a [standard set of results](#results).
+
+This section defines the [parameters](#parameters) accepted by coin selection
+algorithms used in Cardano Wallet, the [results](#results) returned, as well as
+the various [error conditions](#failure-modes) that may occur.
+
+In this section, the terms _coin selection algorithm_ and _coin selection
+function_ will be used interchangeably.
+
+## Parameters
+
+A coin selection function accepts the following parameters:
+
+ 1. #### Requested Output List
+
+    A list of payments to be made to recipient addresses, encoded as a list of
+    [transaction outputs](#transaction-output).
+
+ 2. #### Initial UTxO Set
+
+    A subset of a wallet's [UTxO set](#utxo-set).
+
+    The coin selection algorithm will select entries from within this set in
+    order to cover payments listed in the [requested output
+    list](#requested-output-list).
+
+    Normally, this parameter would be assigned with the complete [UTxO
+    set](#utxo-set) of a wallet, giving the coin selection algorithm access to
+    the total value associated with that wallet.
+
+ 3. #### Maximum Input Count
+
+    An _upper bound_ on the number of UTxO entries that the coin selection
+    algorithm is permitted to select from the [initial UTxO
+    set](#initial-utxo-set).
+
+## Results
+
+A coin selection function produces the following result values:
+
+ 1. #### Coin Selection
+
+    The _coin selection_ is a record with three fields:
+
+      * A set of **_inputs_**, equivalent to a subset of the
+        [initial UTxO set](#initial-utxo-set).
+
+        Represents the value that has been selected from the wallet in order to
+        cover the total payment value.
+
+      * A set of **_outputs_** (see [transaction output](#transaction-output)).
+
+        Represents the set of payments to be made to recipient addresses.
+
+      * A set of **_change values_**, where each change value is simply a
+        [coin amount](#coin-amount).
+
+        Represents the change to be returned to the wallet.
+
+ 2. #### Remaining UTxO Set
+
+    The _remaining UTxO set_ is a subset of the [initial UTxO
+    set](#initial-utxo-set).
+
+    It represents the set of values that remain after the coin selection
+    algorithm has removed values to pay for entries in the [requested output
+    list](#requested-output-list).
+
+    If a coin selection algorithm is applied to the _complete_ UTxO set of a
+    wallet, then the remaining UTxO set represents the _updated_ UTxO set of the
+    wallet.
```

Comment:
I wouldn't put too much emphasis on wallets in this document. A coin selection selects elements from a UTxO set. Period. This can obviously be applied to UTxO management inside a wallet, but that is out-of-scope of the current spec IMO.

Reply:
@KtorZ wrote:
> I wouldn't put too much emphasis on wallets in this document. 

I agree that we should avoid making the descriptions of these algorithms depend on the details of wallets. Though perhaps there is a balance that we can strike here?

Of course, the concept of a coin selection algorithm can be completely divorced from the concept of a wallet. So we take a lot of care, for example, to talk about the `initial UTxO set` rather than the `wallet UTxO set`.

But perhaps it helps to occasionally mention the concept of a wallet, as that is a very common context in which a UTxO appears, and it gives readers a bit of context for how these algorithms are used.

How about the following **alternative wording**:

> In the context of a _wallet_, if a coin selection algorithm is applied to the wallet's _complete_ UTxO set, then the remaining UTxO set represents the _updated_ UTxO set of that wallet.

Reply:
@KtorZ I've now updated this section with the alternative wording indicated above.

See here: 874538b1e28327b8004d9eaef83113d159c019e6.